### DISCUSSION TITLE
Task :caffeine:compileJavaPoetJava FAILED

### DISCUSSION THREAD

**Discussion by manishkrishan** [ben-manes/caffeine#582] [Q&A] [answered]
downloaded 3.0.3 code and to build it with "gradlew build" after downloading dependencies, it failed:
--------------------------------------------------------
Starting a Gradle Daemon (subsequent builds will be faster)

checksum-dependency-plugin is disabled since checksumIgnore property is present
> Task :caffeine:compileJavaPoetJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':caffeine:compileJavaPoetJava'.
> invalid source release: 11
-----------------------------------------------------------

**Comment by ben-manes** [comment, marked as answer]
> invalid source release: 11

This indicates that you do not have JDK11 or higher installed. That is required for 3.x, whereas the v2.dev branch supports JDK8.

Currently we don't have Gradle's support for auto-installing the jdk enabled. That should probably be done as a friendly bootstrap mechanism.

**Reply by manishkrishan** [reply]
thanks, some indication on this in readme will help

**Reply by ben-manes** [reply]
Added the toolchain config to master, so this will set up the jdk automatically. That defaults to 11 if not overridden by `JAVA_VERSION`, which is set by the CI to allow for a matrix build.

**Reply by manishkrishan** [reply]
very quick, thanks
